rotkehlchen/chain/ethereum/modules/lido/decoder.py

```diff
@@ -0,0 +1,526 @@
+
+        action_items = []  # also create an action item for the receive of the wstETH tokens
+        if paired_eth_transfer_event is not None and action_from_event_type is not None:
+            action_items.append(ActionItem(
+                action='transform',
+                from_event_type=action_from_event_type,
+                from_event_subtype=HistoryEventSubType.NONE,
+                asset=A_WSTETH,
+                amount=wsteth_minted_tokens,
+                to_event_subtype=action_to_event_subtype,
+                to_notes=action_to_notes,
+                to_counterparty=CPT_LIDO_ETH,
+                paired_event_data=(paired_eth_transfer_event, True),
+                extra_data={'staked_eth': str(eth_deposited_amount)},
+            ))
+
+        return DecodingOutput(action_items=action_items, matched_counterparty=CPT_LIDO_ETH)
+
+    def _decode_lido_eth_staking_contract(self, context: DecoderContext) -> DecodingOutput:
+        """Decode interactions with stETH ans wstETH contracts"""
+
+        if (
+            context.tx_log.topics[0] not in {
+                STETH_SUBMIT_TOPIC,
+                STETH_TRANSFER_SHARES_TOPIC,
+            }
+        ):
+            return DEFAULT_DECODING_OUTPUT
+
+        if (
+            context.tx_log.topics[0] == STETH_SUBMIT_TOPIC and
+            self.base.any_tracked([
+                hex_or_bytes_to_address(context.tx_log.topics[1]),  # sender address
+            ])
+        ):
+            return self._decode_lido_staking_in_steth(
+                context=context,
+            )
+        elif (  # wsteth supports also the automatic staking with direct ETH transfers
+            context.tx_log.topics[0] == STETH_SUBMIT_TOPIC and
+            hex_or_bytes_to_address(context.tx_log.topics[1]) == wsteth_evm_address  # sender address  # noqa: E501
+
+        ):
+            return self._maybe_decode_lido_staking_with_eth_transfer_to_wsteth(
+                context=context,
+            )
+        elif (  # there is no event log emitted for wrap and unwrapping, using TransferShares topic and ERC20 transfers  # noqa: E501
+            context.tx_log.topics[0] == STETH_TRANSFER_SHARES_TOPIC and
+            self.base.any_tracked([
+                hex_or_bytes_to_address(context.tx_log.topics[1]),  # from
+                hex_or_bytes_to_address(context.tx_log.topics[2]),  # to
+            ]) and
+            wsteth_evm_address in {
+                hex_or_bytes_to_address(context.tx_log.topics[1]),  # from
+                hex_or_bytes_to_address(context.tx_log.topics[2]),  # to
+            }
+        ):
+            return self._decode_steth_wrap_unwrap(
+                context=context,
+            )
+        else:
+            return DEFAULT_DECODING_OUTPUT
+
+    # -- DecoderInterface methods
+
+    def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
+        return {
+            steth_evm_address: (self._decode_lido_eth_staking_contract,),
```

Comment:
Ok, so please propose and educate me, which other address we can use for decoding the following eth submit transaction? https://etherscan.io/tx/0x23a3ee601475424e91bdc0999a780afe57bf37cbcce6d1c09a4dfaaae1765451#eventlog

Reply:
Can't use decoder mapping for this. It's a bad case where our decoders will have to be slow I am afraid. I would like to hear also @yabirgb's view.

But perhaps putting nothing except for a post decoding rule where you just use the existing decoded transfer from 0x.00 to the tracked address and transform this is the most performant thing to do.

Reply:
Ok, using only post decoding it is something I can try, it seems to me it should work, and I could move the logic there. Only difference is that the code would not check anymore that "submit" method was called, but it should work because it seems it is only case where stETH are generated from zero address and transferred to the user address.

Reply:
but at the same time, you said in other comment that post decoding rules should be avoided, right? https://github.com/rotki/rotki/pull/7949#discussion_r1605903531

Reply:
yes it should be avoided so long as other options can work fine. Here they won't.

And address decoder based on a token's address is also pretty bad since it will run for every transfer/approval/other event related to the token.

From the information you shared, so far as I can see with the time I have now it's indeed a case of an event that can only be done via suboptimal decoders :(